ecl/data_source_ecl_network_fic_gateway_v2.go

```diff
@@ -0,0 +1,126 @@
+package ecl
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/nttcom/eclcloud/ecl/network/v2/fic_gateways"
+)
+
+func dataSourceNetworkFICGatewayV2() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkFICGatewayV2Read,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_service_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"qos_option_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tenant_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceNetworkFICGatewayV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkClient, err := config.networkV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL network client: %w", err)
+	}
+
+	listOpts := fic_gateways.ListOpts{}
+
+	if v, ok := d.GetOk("description"); ok {
+		listOpts.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_service_id"); ok {
+		listOpts.FICServiceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_gateway_id"); ok {
+		listOpts.ID = v.(string)
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		listOpts.Name = v.(string)
+	}
+
+	if v, ok := d.GetOk("qos_option_id"); ok {
+		listOpts.QoSOptionID = v.(string)
+	}
+
+	if v, ok := d.GetOk("status"); ok {
+		listOpts.Status = v.(string)
+	}
+
+	if v, ok := d.GetOk("tenant_id"); ok {
+		listOpts.TenantID = v.(string)
+	}
+
+	pages, err := fic_gateways.List(networkClient, listOpts).AllPages()
+	if err != nil {
+		return fmt.Errorf("unable to retrieve fic_gateways: %w", err)
+	}
+
+	allFICGateways, err := fic_gateways.ExtractFICGateways(pages)
+	if err != nil {
+		return fmt.Errorf("unable to extract fic_gateways: %w", err)
+	}
+
+	if len(allFICGateways) < 1 {
+		return fmt.Errorf("Your query returned no results." +
```

Comment:
エラーメッセージが大文字で始まっているので修正してください。108行目も同様です。

Reply:
未修正のようです。
```
if len(gws) < 1 {
		return fmt.Errorf("your query returned no results." +
			" Please change your search criteria and try again.")
	}
```

Reply:
上のエラーですが、最後がドットで終わっているのが問題でした。修正をお願いします。
> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. 
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
直接該当行にコメントしました。